### PR TITLE
Applied TranslationCards, NewTranslationCard and ResumeBookBanner on Home Page 

### PR DESCRIPTION
**Pull request by mXaln**
This change is [Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/301)

### CODE REVIEW

**Review by jsarabia — CHANGES_REQUESTED** [review]
I forgot in one of the recent PRs that we need to include database migrations since the table changed; installing overtop previous installs will cause crashes on startup when importlanguages is run.

This change should be done in:
jvm.workbookapp.persistence.database.DatabaseMigrator
and the Schema version should be incremented to 3

> Reviewed 3 of 27 files at r2.
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/301#-:-Maof_d6B1XaEQTuYBw0:bkrbwf0)* status: 3 of 34 files reviewed, 2 unresolved discussions (waiting on @mXaln)

___
*[common/src/main/kotlin/org/wycliffeassociates/otter/common/persistence/repositories/IAppPreferencesRepository.kt, line 6 at r3](https://reviewable.io/reviews/bible-translation-tools/orature/301#-MaoaCpo1Y-lE7CCOdxZ:-MaoaCpo1Y-lE7CCOdx_:bsjqat4) ([raw file](https://github.com/bible-translation-tools/orature/blob/d142dc0ea21f94432ad0ff87c07ee4b47197a0b3/common/src/main/kotlin/org/wycliffeassociates/otter/common/persistence/repositories/IAppPreferencesRepository.kt#L6)):*
> ```Kotlin
> import io.reactivex.Single
> 
> interface IAppPreferencesRepository {
> ```

Is there a reason we're introducing a new repository? were other preferences built into other repository interfaces?
___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/BookWizardViewModel.kt, line 235 at r3](https://reviewable.io/reviews/bible-translation-tools/orature/301#-MaobPNgCscsHAVvDXgc:-MaobPNgCscsHAVvDXgd:b3wocu9) ([raw file](https://github.com/bible-translation-tools/orature/blob/d142dc0ea21f94432ad0ff87c07ee4b47197a0b3/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/BookWizardViewModel.kt#L235)):*
> ```Kotlin
>             createCheckboxMenuItem(messages["oldTestament"], true, anthologyVisibleProperty) { selected ->
>                 when (selected) {
>                     true -> if (!selectedAnthologies.contains(SlugsEnum.OT)) selectedAnthologies.add(SlugsEnum.OT)
> ```

I'd really like to get rid of SlugsEnum; and make some refactors here. SlugsEnum was always going against our design principles. We want to have content come into Orature, and everything that's needed needs to come from the content. We do not want Orature to know about content and leak content knowledge into code.

Knowing that the ULB has anthologies, and that those anthologies are OT/NT and which books specifically belong to which is bad, and indicates that we handled the import of the Resource Container poorly.

At the time, I really lamented the way that the RC spec handled old testament/new testament. It basically doesn't. Everything about the Resource Container that defines content and what is in it does so by providing keys/slugs as well as the localized display text (ie: ulb and Unlocked Literal Bible, gen and Genesis). The only way that anthologies are "handled" in RCs is by the "categories" list, which only provides the slugs "bible-ot" and "bible-nt". The localized words "Old Testament" and "New Testament" do not exist in the manifest.

That said, to compensate for this, I feel that we should handle this lack of data in exactly ONE place in Orature- at import. We should consider these "Door43 Resource Containers" and bible-ot/bible-nt as keywords as part of their spec. On importing "Door43 Resource Containers" we should handle the adding in of missing data/text as part of the import/data sanitization process, that gets everything into a standard format Orature expects, so that it then treats all content the same.

One significant reason for doing this is due to the fact that the Bible is not the ULB, much like a Rectangle is not a Square. A Square is a type of rectangle, and similarly the ULB is a specific version of the Bible. The AYT is another; which, can be imported by Orature, but is not ULB. the French f10 is another example. This is why I would prefer content discrepancies be handled in content, such that we're not having to add all of these things as enums or writing special cases in code.

This "Door43 Import" should be some kind of implementation of some more generic interface as well. In fact, maybe the Resource Container is one type of Importable format- a generic interface that someone could create an epub importer for, as an example. Or Scripture Burrito.

Let's talk about this on Tuesday; we can see if there's minimal changes we can make now, and push the rest off perhaps to when we refactor import export.

**Review by mXaln — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/301)* status: 3 of 62 files reviewed, 2 unresolved discussions (waiting on @jsarabia)

___
*[common/src/main/kotlin/org/wycliffeassociates/otter/common/persistence/repositories/IAppPreferencesRepository.kt, line 6 at r3](https://reviewable.io/reviews/bible-translation-tools/orature/301#-MaoaCpo1Y-lE7CCOdxZ:-Mb6BCNJ3E8TiNRxcDkv:biyeiue) ([raw file](https://github.com/bible-translation-tools/orature/blob/d142dc0ea21f94432ad0ff87c07ee4b47197a0b3/common/src/main/kotlin/org/wycliffeassociates/otter/common/persistence/repositories/IAppPreferencesRepository.kt#L6)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

Is there a reason we're introducing a new repository? were other preferences built into other repository interfaces?
</blockquote></details>

yes, other preferences were built into AudioPluginRepository
___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/BookWizardViewModel.kt, line 235 at r3](https://reviewable.io/reviews/bible-translation-tools/orature/301#-MaobPNgCscsHAVvDXgc:-Mb6CGv4CUqMWQUitr3P:byb7qkc) ([raw file](https://github.com/bible-translation-tools/orature/blob/d142dc0ea21f94432ad0ff87c07ee4b47197a0b3/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/BookWizardViewModel.kt#L235)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

I'd really like to get rid of SlugsEnum; and make some refactors here. SlugsEnum was always going against our design principles. We want to have content come into Orature, and everything that's needed needs to come from the content. We do not want Orature to know about content and leak content knowledge into code.

Knowing that the ULB has anthologies, and that those anthologies are OT/NT and which books specifically belong to which is bad, and indicates that we handled the import of the Resource Container poorly.

At the time, I really lamented the way that the RC spec handled old testament/new testament. It basically doesn't. Everything about the Resource Container that defines content and what is in it does so by providing keys/slugs as well as the localized display text (ie: ulb and Unlocked Literal Bible, gen and Genesis). The only way that anthologies are "handled" in RCs is by the "categories" list, which only provides the slugs "bible-ot" and "bible-nt". The localized words "Old Testament" and "New Testament" do not exist in the manifest.

That said, to compensate for this, I feel that we should handle this lack of data in exactly ONE place in Orature- at import. We should consider these "Door43 Resource Containers" and bible-ot/bible-nt as keywords as part of their spec. On importing "Door43 Resource Containers" we should handle the adding in of missing data/text as part of the import/data sanitization process, that gets everything into a standard format Orature expects, so that it then treats all content the same.

One significant reason for doing this is due to the fact that the Bible is not the ULB, much like a Rectangle is not a Square. A Square is a type of rectangle, and similarly the ULB is a specific version of the Bible. The AYT is another; which, can be imported by Orature, but is not ULB. the French f10 is another example. This is why I would prefer content discrepancies be handled in content, such that we're not having to add all of these things as enums or writing special cases in code.

This "Door43 Import" should be some kind of implementation of some more generic interface as well. In fact, maybe the Resource Container is one type of Importable format- a generic interface that someone could create an epub importer for, as an example. Or Scripture Burrito.

Let's talk about this on Tuesday; we can see if there's minimal changes we can make now, and push the rest off perhaps to when we refactor import export.
</blockquote></details>

yeah, sounds cool to discuss this

**Review by jsarabia — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/301)* status: 1 of 72 files reviewed, 2 unresolved discussions (waiting on @jsarabia and @mXaln)

___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/AddFilesViewModel.kt, line 112 at r8](https://reviewable.io/reviews/bible-translation-tools/orature/301#-MbCk3Lt40CGCkUzfqk3:-MbCk3Lt40CGCkUzfqk4:b-lek7s4) ([raw file](https://github.com/bible-translation-tools/orature/blob/edf4c8b1f98be865d0a330319c309f149f016581/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/AddFilesViewModel.kt#L112)):*
> ```Kotlin
>                         logger.debug("Error in getSourceMetadata: $rc")
>                     }
>                     .onErrorReturnItem(null)
> ```

This will probably crash, since nulls can't be emitted in rx streams?
___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/HomePageViewModel.kt, line 122 at r2](https://reviewable.io/reviews/bible-translation-tools/orature/301#-MbCiy-W5DAyoZ-ikpO0:-MbCiy-X-jsm8fbD-xOf:bz8vo09) ([raw file](https://github.com/bible-translation-tools/orature/blob/1d23aec076f91d0454ab8e93245e3d42440a30f0/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/HomePageViewModel.kt#L122)):*
> ```Kotlin
>     }
> 
>     private fun translationCardModelSorter(): Comparator<TranslationCardModel> {
> ```

Maybe we should just use translationCardModelComparator instead of sorter

**Review by mXaln — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/301)* status: 1 of 73 files reviewed, 2 unresolved discussions (waiting on @jsarabia)

___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/AddFilesViewModel.kt, line 112 at r8](https://reviewable.io/reviews/bible-translation-tools/orature/301#-MbCk3Lt40CGCkUzfqk3:-MbD-Tzr2NJz6HEXkvwV:bien1bf) ([raw file](https://github.com/bible-translation-tools/orature/blob/edf4c8b1f98be865d0a330319c309f149f016581/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/AddFilesViewModel.kt#L112)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

This will probably crash, since nulls can't be emitted in rx streams?
</blockquote></details>

I had to change the observable to Maybe in order to complete on error
___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/HomePageViewModel.kt, line 122 at r2](https://reviewable.io/reviews/bible-translation-tools/orature/301#-MbCiy-W5DAyoZ-ikpO0:-MbCotSq6QoNN4iSa-JG:b-896fix) ([raw file](https://github.com/bible-translation-tools/orature/blob/1d23aec076f91d0454ab8e93245e3d42440a30f0/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/HomePageViewModel.kt#L122)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

Maybe we should just use translationCardModelComparator instead of sorter
</blockquote></details>

Done.

**Review by jsarabia — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r1, 19 of 27 files at r2, 6 of 8 files at r3, 22 of 28 files at r4, 7 of 22 files at r7, 1 of 1 files at r8, 2 of 3 files at r9.
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/301)* status: 58 of 73 files reviewed, 1 unresolved discussion (waiting on @jsarabia and @mXaln)

___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/TranslationViewModel.kt, line 38 at r9](https://reviewable.io/reviews/bible-translation-tools/orature/301#-MbGpin_9BDYL4hGvXJ7:-MbGpin_9BDYL4hGvXJ8:b5kjsuf) ([raw file](https://github.com/bible-translation-tools/orature/blob/92ba82026ab70285753b52f957f2c68708b9c506/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/TranslationViewModel.kt#L38)):*
> ```Kotlin
>     val showProgressProperty = SimpleBooleanProperty(false)
> 
>     private val sourceCollections = mutableListOf<ResourceMetadata>()
> ```

I would probably name these sourceResources, since they're the metadata, not collection objects

**Review by mXaln — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/301)* status: 49 of 87 files reviewed, 1 unresolved discussion (waiting on @jsarabia)

___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/TranslationViewModel.kt, line 38 at r9](https://reviewable.io/reviews/bible-translation-tools/orature/301#-MbGpin_9BDYL4hGvXJ7:-MbGwbwnDraNk7dR-r7U:b-896fix) ([raw file](https://github.com/bible-translation-tools/orature/blob/92ba82026ab70285753b52f957f2c68708b9c506/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/TranslationViewModel.kt#L38)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

I would probably name these sourceResources, since they're the metadata, not collection objects
</blockquote></details>

Done.

File: jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/HomePageViewModel.kt [hunk 1]
```diff
@@ -42,6 +42,8 @@ class HomePageViewModel : ViewModel() {
     val translationModels = observableListOf<TranslationCardModel>()
     val resumeBookProperty = SimpleObjectProperty<Workbook>()
 
+    private val noResumableProject = -1
```
Comment (jsarabia) [inline jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/HomePageViewModel.kt:45]: This should be a const at the top of the file, and all caps with underscores NO_RESUMABLE_PROJECT

**Pull request merged**